src/mon/OSDMonitor.cc

```diff
@@ -4586,6 +4745,105 @@ void OSDMonitor::tick()
   if (do_propose ||
       !pending_inc.new_pg_temp.empty())  // also propose if we adjusted pg_temp
     propose_pending();
+
+  {
+    std::lock_guard l(balancer_lock);
+    if (ceph_using_tcmalloc() && mon_memory_autotune && pcm != nullptr) {
+      _adjust_cache_ratios();
+      pcm->balance();
```

Comment:
If running balance() and tune_memory() in the same loop, I'd suggest calling tune_memory first.  That way you're figuring out what the aggregate cache size should be and then balancing based on it rather than balancing on what the old aggregate cache size is.

Reply:
Done.